onnxruntime/python/tools/transformers/shape_infer_helper.py

```diff
@@ -0,0 +1,115 @@
+import os
+import sys
+
+# In ORT Package the symbolic_shape_infer.py is in ../tools
+file_path = os.path.dirname(__file__)
+if os.path.exists(os.path.join(file_path, "../tools/symbolic_shape_infer.py")):
+    sys.path.append(os.path.join(file_path, '../tools'))
+else:
+    sys.path.append(os.path.join(file_path, '..'))
+from symbolic_shape_infer import *
+
+import onnx
+
+
+class SymbolicShapeInferenceHelper(SymbolicShapeInference):
+    def __init__(self, model, verbose=0, int_max=2**31 - 1, auto_merge=True, guess_output_rank=False):
+        super().__init__(int_max, auto_merge, guess_output_rank, verbose)
+        self.model_ = onnx.ModelProto()
+        self.model_.CopyFrom(model)
+
+        self.dispatcher_['Gelu'] = self._infer_Gelu
```

Comment:
Maybe we shall add shape inference code for these contrib ops to symbolic_shape_infer.py

Reply:
Agree. I had thoughts about a more generic domain and opset version support in symbolic_shape_inference before, but didn't have a use case then. Feel free to change that script as if you own it.

---
In reply to: [589870429](https://github.com/microsoft/onnxruntime/pull/6899#discussion_r589870429) [](ancestors = 589870429)